WebsiteGreetz/src/test/java/WebsiteTest.java

```
@@ -0,0 +1,81 @@
+        driver.get("https://www.greetz.nl/mygreetz/favorieten");
+        Thread.sleep(5000);
+        if(driver.findElement(By.xpath("//button[@class='b-icon_medium b-icon-delete']")).isDisplayed()) {
+            WebElement deleteConfirmationButton = driver.findElement(By.xpath("//button[@class='b-icon_medium b-icon-delete']"));
+            deleteConfirmationButton.click();
+            WebElement okDelete = driver.findElement(By.xpath("//span[contains(text(), 'Ok')]"));
+            okDelete.click();
+            }
+        driver.get("https://www.greetz.nl/ballonnen/denken-aan");
+        Thread.sleep(3000);
+        WebElement targetItem = driver.findElement(By.xpath("//a[@data-id='1142804226']"));
+
+        WebElement targetElementFavouritebutton = targetItem.findElement(By.xpath("//parent::div//child::a[@class='b-products-grid__item-action']"));
+        targetElementFavouritebutton.click();
+
+        WebElement sideMenu = driver.findElement(By.xpath("//i[@class='page-header__navigation-item-icon b-icon b-icon-hamburger']"));
+        sideMenu.click();
+
+        WebElement favourite = driver.findElement(By.xpath("//span[@class='b-list--item-subject' and contains(text(), 'Favorieten')]"));
+        favourite.click();
+
+        int countOfFavorites = 0;
+        Assert.assertTrue(findFavorite(countOfFavorites));
+    }
+
+    @Test
+    public void priceForTwoCards() throws InterruptedException {
+        driver.get("https://www.greetz.nl/kaarten/denken-aan");
+        Thread.sleep(5000);
+        WebElement card = driver.findElement(By.xpath("//div[@class='b-products-grid__item'][2]"));
+        card.click();
+
+
+        Thread.sleep(5000);
+        WebElement input = driver.findElement(By.xpath("//input[@name='amount']"));
+        input.clear();
+        input.sendKeys("2");
+        Thread.sleep(2000);
+        Assert.assertTrue(driver.findElement(By.xpath("//div[@class='price-total' and contains(text(), '5,90')]")).isDisplayed());
```

Hardcoded values should not be used in the test. The price can be changed frequently. Your test will fail even more frequently :) . Also because you used index for navigating to the element. And the index itself can be changed. So, you never know whether your test uses the same element or not.

---

Can you please explain it once more? I don't understand what I need to change.